src/team_lib/mhba/xccl_mhba_collective.c

```diff
@@ -547,7 +571,6 @@ xccl_status_t xccl_mhba_alltoall_init(xccl_coll_op_args_t  *coll_args,
     request->seq_num = team->sequence_number;
     xccl_mhba_debug("Seq num is %d", request->seq_num);
     team->sequence_number++;
-
     for (i = 0; i < n_tasks; i++) {
         request->tasks[i].req = request;
         xccl_coll_task_init(&request->tasks[i].super);
```

Comment:
need to reset mkey_cache_flag + add check to reset seq_num if reached MAX

Reply:
don't need to: mkey_cache_flag is initialized each time (former state is not re-used) so, not need to reset to 0.
and in wraparound is find, it'll start from 0.

Reply:
regarding seq_num check, look in 104